blockchain/validators/src/metadata/key_range/config.rs

```diff
@@ -38,13 +37,17 @@ impl Config {
     pub fn load(path: impl AsRef<Path>) -> Result<Self, ParseError> {
         let path = path.as_ref();
         let bytes = fs::read(path)?;
-        let mut config: Config = match path.extension().and_then(OsStr::to_str) {
-            Some("toml") => Ok(toml::from_slice(&bytes)?),
-            Some("json") => Ok(serde_json::from_slice(&bytes)?),
-            _ => Err(ParseError::UnrecognizedExtension(path.into())),
-        }?;
-        config.base_path = path.parent().map(Into::into);
-        Ok(config)
+        if let Ok(mut config) = serde_json::from_slice(&bytes): Result<Self, _> {
+            config.base_path = path.parent().map(Into::into);
+            Ok(config)
+        } else if let Ok(mut config) = toml::from_slice(&bytes): Result<Self, _> {
+            config.base_path = path.parent().map(Into::into);
+            Ok(config)
+        } else {
+            Err(ParseError::UnsupportedFileFormat(
+                path.to_string_lossy().into_owned(),
+            ))
```

Comment:
This approach would fail with only an "Unsupported file" error if we get an otherwise valid TOML or JSON, e.g. trailing comma in JSON, mismatched quotes or other delimiter in TOML or JSON